packages/server-common/src/handlers/utils.ts

```diff
@@ -26,7 +26,8 @@ export const createProcessingFunction =
     type: Type,
     logger: Logger,
     filterFunction: (item: T['data']) => boolean = () => true,
-    addTagsFunction?: (item: T['data']) => T['data'] & { _tags: string[] },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    addTagsFunction?: (item: any) => SavePayload['data'],
```

Comment:
this is strange, why are you having to do this?

Reply:
Actually I think I can undo this change. It was necessary when the data provider was returning the old UserResponse and I was calling the function to convert into the UserListItemResponse here. So the function couldn't be types as `(item: T['data']) => T['data'] & { _tags: string[] }` because the input would be `UserResponse` and output `UserListItemResponse`. And the input now could be anything. 

I was calling `toUserListItem` in `algolia-index-team-users-handler.ts` but now that the data provider sends exactly UserListItemResponse. I could just remove it.